pkg/cloud/cloud.go

```diff
@@ -33,9 +33,15 @@ func New(region, kmsEndpoint string, qps, burst int) (*AWSKMS, error) {
 		CredentialsChainVerboseErrors: aws.Bool(true),
 		Endpoint:                      aws.String(kmsEndpoint),
 	}
+
+	sess, err := session.NewSession()
```

Comment:
is this similar to pattern followed in https://github.com/aws/aws-sdk-go/blob/main/aws/session/custom_ca_bundle_test.go#L52-L59 ?

Reply:
That was the previous setup and it failed

Reply:
Does that mean `sess.HttpClient` overides `cfg.HttpClient` ? From https://github.com/aws/aws-sdk-go/blob/main/aws/session/session.go#L593 i see the CA bundle being used for the `cfg.HttpClient`

Reply:
is it possible to add a unit test similar to https://github.com/aws/aws-sdk-go/blob/main/aws/session/custom_ca_bundle_test.go#L52-L59 to make sure a future change doesn't revert this back?

Reply:
is it possible to add a unit test similar to https://github.com/aws/aws-sdk-go/blob/main/aws/session/custom_ca_bundle_test.go#L52-L59 to make sure a future change doesn't revert this back?

Reply:
ok nvm my previous qs.. if cfg is not provided then the session.HttpClient is used https://github.com/aws/aws-sdk-go/blob/9e261711fdee3b565ef0cb4f94a28610decaa0e0/aws/config.go#L505-L507 
Wondering how changing the HttpClient frm cfg to session helps.

Reply:
Looking at the docs for [Custom CA Bundle](https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#:~:text=Custom%20CA%20Bundle) it states:
> The Session option CustomCABundle is also available when creating sessions to also enable this feature. CustomCABundle session option field has priority over the AWS_CA_BUNDLE environment variable, and will be used if both are set.
> Setting a custom HTTPClient in the aws.Config options will override this setting. To use this option and custom HTTP client, the HTTP client needs to be provided when creating the session. Not the service client.

This code LGTM

Reply:
Makes sense to me now. I would prefer if we could attempt a unit test instead of testing it manually on a cluster

Reply:
updated with unit tests @jyotimahapatra